airflow/providers/google/cloud/transfers/azure_fileshare_to_gcs.py

```diff
@@ -0,0 +1,217 @@
+        self.prefix = prefix
+        self.wasb_conn_id = wasb_conn_id
+
+        if dest_gcs_conn_id:
+            warnings.warn(
+                "The dest_gcs_conn_id parameter has been deprecated. You should pass "
+                "the gcp_conn_id parameter.",
+                DeprecationWarning,
+                stacklevel=3,
+            )
+            gcp_conn_id = dest_gcs_conn_id
+
+        self.gcp_conn_id = gcp_conn_id
+        self.dest_gcs = dest_gcs
+        self.delegate_to = delegate_to
+        self.replace = replace
+        self.gzip = gzip
+        self.google_impersonation_chain = google_impersonation_chain
+
+        if dest_gcs and not self._gcs_object_is_directory(self.dest_gcs):
+            self.log.info(
+                'Destination Google Cloud Storage path is not a valid '
+                '"directory", define a path that ends with a slash "/" or '
+                'leave it empty for the root of the bucket.'
+            )
+            raise AirflowException(
+                'The destination Google Cloud Storage path ' 'must end with a slash "/" or be empty.'
+            )
+
+    def execute(self, context):
+        azure_fileshare_hook = AzureFileShareHook(self.wasb_conn_id)
+        files = azure_fileshare_hook.list_files(
+            share_name=self.share_name, directory_name=self.directory_name
+        )
+
+        gcs_hook = GCSHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.google_impersonation_chain,
+        )
+
+        # pylint: disable=too-many-nested-blocks
+        if not self.replace:
+            # if we are not replacing -> list all files in the GCS bucket
+            # and only keep those files which are present in
+            # S3 and not in Google Cloud Storage
+            bucket_name, object_prefix = _parse_gcs_url(self.dest_gcs)
+            existing_files_prefixed = gcs_hook.list(bucket_name, prefix=object_prefix)
+
+            existing_files = []
+
+            if existing_files_prefixed:
+                # Remove the object prefix itself, an empty directory was found
+                if object_prefix in existing_files_prefixed:
+                    existing_files_prefixed.remove(object_prefix)
+
+                # Remove the object prefix from all object string paths
+                for f in existing_files_prefixed:
+                    if f.startswith(object_prefix):
+                        existing_files.append(f[len(object_prefix) :])
+                    else:
+                        existing_files.append(f)
+
+            files = list(set(files) - set(existing_files))
+            if len(files) > 0:
+                self.log.info('%s files are going to be synced: %s.', len(files), files)
+            else:
+                self.log.info('There are no new files to sync. Have a nice day!')
+
+            if files:
+                for file in files:
```

Comment:
```suggestion

                for file in files:
```
If files are empty then there will be nothing to iterate

Reply:
I believe that main purpose of this `if` is to make it possible to decide which message should be logged - either "All done (...)" or "In sync, no files needed (...)" - however I agree that its a good idea to remove this `if` from here and place it directly over the messages.